nf_core/components/create.py

```diff
@@ -411,3 +430,160 @@ def _get_username(self):
                 f"[violet]GitHub Username:[/]{' (@author)' if author_default is None else ''}",
                 default=author_default,
             )
+
+    def _copy_old_files(self, component_old_path):
+        """Copy files from old module to new module"""
+        log.debug("Copying original main.nf file")
+        shutil.copyfile(component_old_path / "main.nf", self.file_paths[self.component_type + "/main.nf"])
```

Comment:
why not `self.component_dir` as the target for the main.nf?

Reply:
oh, yes that's bad. maybe if we initialize the type as string for this variable it becomes clearer when we make this mistake? other option: rename it to `component_dir_name`